api/jobset/v1alpha2/jobset_types.go

```diff
@@ -27,6 +27,11 @@ const (
 	JobKey                string = "jobset.sigs.k8s.io/job-key"
 	JobNameKey            string = "job-name" // TODO(#26): Migrate to the fully qualified label name.
 	ExclusiveKey          string = "alpha.jobset.sigs.k8s.io/exclusive-topology"
+	// NodeSelectorStrategyKey is an annotation that acts as a flag, the value does not matter.
+	// If set, the JobSet controller will automatically inject nodeSelectors for the JobSetNameKey label to
+	// ensure exclusive job placement per topology, instead of injecting pod affinity/anti-affinites for this.
+	// The user must add the JobSet name node label to the desired topologies separately.
+	NodeSelectorStrategyKey string = "alpha.jobset.sigs.k8s.io/node-selector"
```

Comment:
lets do that in an annotation, I am not sure if we will carry this long term

Reply:
This is used as an annotation in the controller - can you clarify what you mean? Do you mean we shouldn't include it in the API types/definitions file since we may not carry it long term?

Reply:
ah, sorry, tired brain :)